tests/framework/clients/object.go

```diff
@@ -90,6 +90,49 @@ spec:
 	return k8sh.CreateExternalRGWService(namespace, storeName)
 }
 
+func (ro *ObjectOperation) ObjectDelete(namespace, storeName string, replicaCount int32, callAPI bool, k8sh *utils.K8sHelper) error {
+	store := model.ObjectStore{Name: storeName, Gateway: model.Gateway{Instances: replicaCount, Port: model.RGWPort}}
+	store.DataConfig.ReplicatedConfig.Size = 1
+	store.MetadataConfig.ReplicatedConfig.Size = 1
+	if callAPI {
+		logger.Infof("Deleting  the object store via the rest API")
+		if err := ro.restClient.DeleteObjectStore(storeName); err != nil {
+			return fmt.Errorf("failed to delete rest api object store. %+v", err)
+		}
+	} else {
+		kind := "ObjectStore"
+		if !k8sh.VersionAtLeast("1.7.0") {
+			kind = "Objectstore"
+		}
+		logger.Infof("Deleting the object store via CRD")
+		storeSpec := fmt.Sprintf(`apiVersion: rook.io/v1alpha1
+kind: %s
```

Comment:
why not use the client-go to create this manifest? Instead of a string

Reply:
I thought it would be easier to use Strings instead of client-go becuse it is easier to update the string whenever there is a update. All the yaml manifest we use in integration tests are paramaterized copies of yamls under /cluster/examples/kubernetes.  So when ever we update  yamls under  /cluster/examples/kubernetes, it would be a fairly quick update on integration tests.  That being said, we want to eventually update the integration test framework to just use client-go and remove all refereenes to kubeclt or string manifests. But that is out of scope for this ticket and it should be handled in a separately.